x2: # That ends with 0.9999 are not investigated yet
  ros__parameters:
    m: [0.7, 2.14, 0.64, 0.47, 1.25, 10.3] # masses of upper thigh, lower thigh, upper shank, lower shank, foot, backpack [kg]  todo 9.9
    #m: [1.0, 2.5, 1.0, 0.8, 0.0, 10.3] # masses of upper thigh, lower thigh, upper shank, lower shank, foot, backpack [kg]  todo 9.9
    l: [0.4, 0.35, 0.4, 0.35] # length of left thigh, left shank, right thigh, right shank [kg]
    s: [0.0811, 0.05564, 0.09080, 0.0562, 0.08834, 0.25, 0.1] # hip to upper thigh COM, knee to lower thigh COM,  # todo calculate and fix last 2 parameters [m]
    # knee to upper shank COM, ankle to lower shank COM, ankle to foot COM, backpack COM to hip parallel to backpack link, backpack COM to hip vertical to backpack link [m]
    I: [0.002197, 0.005901, 0.002346, 0.0007941, 0.004441, 0.2] # mass moment of inertia of upper thigh, lower thigh, upper shank, lower shank, foot, backpack [kg.m^2] #todo calculate and fix backpack inertia
    G: [1.9, 1.9, 1.9, 1.9] # apparent rotor inertias (I_rotor*reduction_ratio^2) [kg. m^2]
    c0: [5.01, 3.26, 4.30, 4.45] # viscous fric constant of joints [N.s] [5.01, 3.26, 4.30, 4.45]
    #c0: [4.01, 1.68, 3.30, 2.225] # viscous fric constant of joints [N.s] [5.01, 3.26, 4.30, 4.45]
    c1: [4.58, 4.67, 3.25, 5.16] # coulomb friction const of joints [N.m] [4.58, 4.67, 3.25, 5.16]
    max_torque: 81 # maximum allowable torque [Nm] If exceeds, robot will be disabled
    #  max_power: 95
    max_velocity: 4.5 # maximum allowable speed [rad/s] If exceeds, robot will be disabled
    position_limit: # [deg]
      hip_max: 120
      hip_min: -40
      knee_max: 0
      knee_min: -120
    # strain gauge filter parameters for 2nd order Butterworth
    filter_coeff_a: [1.0, 2.0, 1.0]
    filter_coeff_b: [124.059, 214.679, -94.619]
    # strain gauge scales
    #strain_gauge_scale: [-0.019193478, -0.016627119, 0.015766071, 0.01308]
    strain_gauge_scale: [1.0, 1.0, 1.0, 1.0]